{{ cookiecutter.formal_name }}/app/src/main/java/org/beeware/android/MainActivity.java

```diff
@@ -158,8 +159,25 @@ private void startPython(Map<String, File> paths) throws Exception {
         String pythonHome = paths.get("stdlib").getAbsolutePath();
         this.setPythonEnvVars(pythonHome);
 
+        Log.d(TAG, "Computing Python version.");
+        // Compute Python version number so that we can make sure it's first on sys.path when we
+        // configure sys.path with Python.init().
+        String[] libpythons = new File(this.getApplicationInfo().nativeLibraryDir).list(
+                new FilenameFilter() {
+                    @Override
+                    public boolean accept(File file, String s) {
+                        return s.startsWith("libpython") && s.endsWith(".so");
+                    }
+                }
+        );
+        if (libpythons.length == 0) {
+            throw new Exception("Unable to compute Python version");
+        }
+        String pythonVersion = libpythons[0].replace("libpython", "").replaceAll("m*.so", "");
```

Comment:
Will this work for Python3.8, which doesn't have the `m` suffix?

Reply:
Yup -- it's a regex, so the `m*` permits the absence of `m`. I tested with all 3 versions.

Reply:
Ah - right - I thought it was a glob.